tools/io-trace-explorer/io_trace.cc

```diff
@@ -0,0 +1,415 @@
+        if (last.timestamp <= item.timestamp)
+            t->_operations.push_back(item);
+        else
+            t->_operations.insert(
+              std::upper_bound(t->_operations.begin(), t->_operations.end(), item), item);
+    }
+}
+
+/*
+ * io_trace_collection::load_from_file --
+ *     Load a collection from file.
+ */
+void
+io_trace_collection::load_from_file(const char *file)
+{
+    FILE *f = fopen(file, "r");
+    if (!f)
+        throw std::runtime_error(
+          std::string("Error opening file \"") + file + "\": " + strerror(errno));
+
+    try {
+        char buf[256];
+        if (fgets(buf, sizeof(buf), f)) {
+            rewind(f);
+            /* Guess the file type from the first line */
+            if (isdigit(buf[0]))
+                load_from_file_blkparse(f);
+            else if (buf[0] == '[')
+                load_from_file_wt_logs(f);
+            else
+                throw std::runtime_error("Unsupported file type");
+        }
+    } catch (std::exception &e) {
+        (void)fclose(f);
+        throw std::runtime_error(std::string("Error while loading \"") + file + "\": " + e.what());
+    }
+
+    if (fclose(f) != 0)
+        throw std::runtime_error(
+          std::string("Error closing file \"") + file + "\": " + strerror(errno));
+}
+
+/*
+ * io_trace_collection::load_from_file_blkparse --
+ *     Load from the blkparse output, using default settings, optionally with the -t argument.
+ */
+
+void
+io_trace_collection::load_from_file_blkparse(FILE *f)
+{
+    /*
+     * Sample output with the default settings from "blkparse -t":
+     *
+     * 259,1   14        2     0.001744322     0  C  WS 365166592 + 8 (  561300) [0]
+     * 259,1    6       46     0.196381562 29456  D  WS 683941352 + 512 (196381562) [wtperf]
+     *
+     * The fields are:
+     *   0) Device
+     *   1) CPU number
+     *   2) Sequence number (within the given CPU)
+     *   3) Timestamp
+     *   4) PID
+     *   5) Action (e.g., D = issue, C = completion)
+     *   6) The RWBS field (R = read, W = write, B = barrier, S = synchronous)
+     *   7) Block offset (using 512-byte blocks)
+     *   8) Always "+"
+     *   9) Block length
+     *  10) Duration in (), in microseconds (enabled via the "-t" command-line argument)
+     *  11) Process name and potentally other process-relevant information in []
+     *
+     * Note that actions other than C and D can carry different extra information on fields 7
+     * and up.
+     */
+
+    char buf[256];
+    unsigned lines = 0;
+    while (fgets(buf, sizeof(buf), f)) {
+        lines++;
+        if (strncmp(buf, "CPU0", 4) == 0)
+            break;
+
+        /* Tokenize the line. */
+
+        int count = 0;
+        char *parts[16];
+        char *next(buf), *p;
+        while ((p = strsep(&next, " \t\n\r")) != NULL) {
+            if (*p == '\0')
+                continue;
+            parts[count++] = p;
+            if (count >= 16)
+                break;
```

Comment:
Maybe use  WT_ELEMENTS(parts) or use the same sizeof trick to avoid having 16 in two places.  (Or use const size value).